torch_sim/runners.py

```diff
@@ -355,29 +369,32 @@ def convergence_fn(
     return convergence_fn
 
 
-def optimize(  # noqa: C901
+def optimize[T: OptimState](  # noqa: C901, PLR0915
     system: StateLike,
     model: ModelInterface,
     *,
-    optimizer: Callable,
-    convergence_fn: Callable | None = None,
+    optimizer: OptimFlavor | tuple[Callable[..., T], Callable[..., T]],
+    cell_filter: "CellFilter | CellFilterFuncs | None" = None,
+    convergence_fn: Callable[[T, torch.Tensor | None], torch.Tensor] | None = None,
```

Comment:
Can't cell_filter just be another `init_kwarg`? Why should it be promoted to the top level API, do all optimizers support it?